frontend/packages/pipelines-plugin/src/components/pipelineruns/detail-page-tabs/PipelineRunVisualization.tsx

```diff
@@ -1,17 +1,33 @@
 import * as React from 'react';
 import { LoadingInline } from '@console/internal/components/utils';
+import { k8sGet } from '@console/internal/module/k8s';
 import PipelineVisualization from '../../pipelines/detail-page-tabs/pipeline-details/PipelineVisualization';
-import { PipelineRunKind } from '../../../types';
-import { getPipelineFromPipelineRun } from '../../../utils/pipeline-augment';
-
+import { PipelineKind, PipelineRunKind } from '../../../types';
 import './PipelineRunVisualization.scss';
+import { getPipelineFromPipelineRun } from '../../../utils/pipeline-augment';
+import { PipelineModel } from '../../../models';
 
 type PipelineRunVisualizationProps = {
   pipelineRun: PipelineRunKind;
 };
 
 const PipelineRunVisualization: React.FC<PipelineRunVisualizationProps> = ({ pipelineRun }) => {
-  const pipeline = getPipelineFromPipelineRun(pipelineRun);
+  const [pipeline, setPipeline] = React.useState<PipelineKind>(null);
+  React.useEffect(() => {
+    const pipelineFromPipelineRun = getPipelineFromPipelineRun(pipelineRun);
+    if (!pipelineFromPipelineRun) {
+      if (pipelineRun.spec.pipelineRef?.name) {
+        const pipelineName = pipelineRun.spec.pipelineRef.name;
+        k8sGet(PipelineModel, pipelineName, pipelineRun.metadata.namespace)
+          .then((newPipeline: PipelineKind) => {
+            setPipeline(newPipeline);
+          })
+          .catch(() => {});
+      }
+    } else {
+      setPipeline(pipelineFromPipelineRun);
+    }
+  }, [pipelineRun]);
   if (!pipeline) {
```

Comment:
What if it's using `spec.pipelineSpec`? You missed my use-case I gave you in DMs around handling that. Perhaps just add that use case to `getPipelineFromPipelineRun` so it tries `status.pipelineSpec` then falls back on `spec.pipelineSpec` and then `null`. Which we can try to fetch the Pipeline.

If it falls all of these things, we need to just hide the visualization... perhaps add another `useState` and when it fails `if (pipelineRun.spec.pipelineRef?.name)`, we just hide the visualization instead of indefinitely spinning.

Reply:
done